manticore/ethereum.py

```diff
@@ -238,6 +242,30 @@ def will_evm_execute_instruction_callback(self, state, instruction, arguments):
             self.add_finding_here(state, 'Reachable SELFDESTRUCT')
 
 
+class DetectEtherLeak(Detector):
+    def will_evm_execute_instruction_callback(self, state, instruction, arguments):
+        if instruction.semantics == 'CALL':
+            dest_address = arguments[1]
+            sent_value = arguments[2]
+            msg_sender = state.platform.current_vm.caller
+
+            # If nothing can be transferred out, we don't care
+            if not state.can_be_true(sent_value > 0):
+                return
+
+            if issymbolic(dest_address):
```

Comment:
This is good enough. Just as fyi what I do is to check that the symbolic value effectivelly has at least 2 solutions. slower.
```
            if issymbolic(address):
                possible_addresses = state.solve_n(address, 2)
                if len(possible_addresses) > 1:
                    self.add_finding_here(state, "Dellegatecall to user controlled address")
```
The question is: What's user controlled?

Reply:
it's slower, but this is not hot code, so it should be fine. i like it.

for the purposes here i think we can define user controlled as : symbolic address

Reply:
`
x = input()
require (x == concrete)
do_stuff(x)   <<<<  x is concrete or symbolic/ x is controlled or not?
`

Reply:
uh i see. you're right, in this case, i could consider it to be effectively concrete. reworking